client/a8c-for-agencies/components/a4a-agency-approval-notice/index.tsx

```diff
@@ -0,0 +1,90 @@
+import { useTranslate } from 'i18n-calypso';
+import LayoutBanner from 'calypso/a8c-for-agencies/components/layout/banner';
+import { useDispatch, useSelector } from 'calypso/state';
+import { getActiveAgency } from 'calypso/state/a8c-for-agencies/agency/selectors';
+import { ApprovalStatus } from 'calypso/state/a8c-for-agencies/types';
+import { savePreference } from 'calypso/state/preferences/actions';
+import { getPreference } from 'calypso/state/preferences/selectors';
+
+import './style.scss';
+
+const AGENCY_APPROVAL_DISMISS_PREFERENCE = 'a4a-agency-approval-notice-dismissed';
+
+const A4AAgencyApprovalNotice = () => {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const agency = useSelector( getActiveAgency );
+
+	const dismissNotice = () => {
+		dispatch( savePreference( AGENCY_APPROVAL_DISMISS_PREFERENCE, true ) );
+	};
+
+	const isDismissed = useSelector( ( state ) =>
+		getPreference( state, AGENCY_APPROVAL_DISMISS_PREFERENCE )
+	);
+
+	if ( isDismissed ) {
+		return null;
+	}
+
+	// If approved, only show banner for 1 week after signup
+	if (
+		agency?.approval_status === ApprovalStatus.APPROVED &&
+		agency?.created_at &&
+		new Date( agency.created_at ) < new Date( Date.now() - 7 * 24 * 60 * 60 * 1000 ) // 7 days
+	) {
+		return null;
+	}
+
+	if ( ! agency?.approval_status ) {
+		return null;
+	}
```

Comment:
I think we can exclude this condition, as the following line covers this already. What do you think?

```
	const bannerDetails = availableBannerDetails[ agency?.approval_status ];

	if ( ! bannerDetails ) {
		return null;
	}
```

Reply:
If I just remove that check it gives a linting error: `Type 'undefined' cannot be used as an index type.`
So I've moved it to be part of the `bannerDetails` definition to hopefully make it more concise.